test/e2e/metrics_e2e_test.go

```diff
@@ -61,15 +69,46 @@ var _ = Describe("Metrics", func() {
 		cleanupCSV()
 
 		// Verify that when the csv has been deleted, it deletes the corresponding CSV metrics
-		Expect(getMetricsFromPod(c, getOLMPod(c), "8081")).ToNot(And(
+		Expect(getMetricsFromPod(c, getPodWithLabel(c, "app=olm-operator"), "8081")).ToNot(And(
 			ContainSubstring("csv_abnormal{name=\"%s\"", failingCSV.Name),
 			ContainSubstring("csv_succeeded{name=\"%s\"", failingCSV.Name),
 		))
+
+		subscriptionCleanup, subscription := createSubscriptionWithInstalledCSV(GinkgoT(), crc, testNamespace, "metric-subscription", testPackageName, stableChannel, installedCSV, v1alpha1.ApprovalManual)
+
+		// Verify metrics have been emitted for subscription
+		Expect(getMetricsFromPod(c, getPodWithLabel(c, "app=catalog-operator"), "8081")).To(And(
```

Comment:
Should these checks use the `Eventually` matcher instead? If I'm reading this correctly, if the metric isn't emitted on time, then the test will fail.